src/pip/_internal/operations/install/wheel.py

```diff
@@ -371,6 +373,9 @@ def save(self) -> None:
                     blocksize = min(zipinfo.file_size, 1024 * 1024)
                     shutil.copyfileobj(f, dest, blocksize)
 
+        mtime = datetime(*zipinfo.date_time).timestamp()
+        os.utime(self.dest_path, (mtime, mtime))
```

Comment:
Perhaps this belongs in a new helper, I'll leave that to maintainer discretion.

Additionally, I can see the rationale behind preserving the existing `atime` here and modifying _only_ the `mtime` of the file, but it requires the addition of `os.stat()` (alas, it does not seem to be possible to pass `NULL` to the underlying call to `utime` to express _"use current time"_)